Sources/CircularProgress/util.swift

```diff
@@ -276,6 +276,21 @@ final class ProgressCircleShapeLayer: CAShapeLayer {
 	}
 }
 
+/**
+Shows the indeterminate state, when it's activated.
+
+It draws part of a circle that gets animated into a looping motion around its core.
+*/
+final class IndeterminateShapeLayer: CAShapeLayer {
+	convenience init(radius: Double, center: CGPoint) {
+		self.init()
+		fillColor = nil
+		path = NSBezierPath.circle(radius: radius, center: bounds.center, startAngle: 270).cgPath
+		anchorPoint = CGPoint(x: 0.5, y: 0.5)
+		position = center
+	}
```

Comment:
@boyvanamstel Would it make sense for this to have a `.isAnimating` property that can start stop the animation?

Then we could move the

```swift
if indeterminateCircle.animation(forKey: "rotate") == nil {
	indeterminateCircle.add(CABasicAnimation.rotate, forKey: "rotate")
}
```

logic here and just do `indeterminateCircle.isAnimating = true` in the `updateColors()` method.

Reply:
It would make it a bit more transparent and easier to handle other animations there later. Might be worth the extra code.